### PR TITLE
#36 Converted AcceptedAgreements to H5

### PR DESCRIPTION
**Pull request by BenGalewsky**
Converted AcceptedAgreements to Hibernate 5
- Created table name in seed.sql
- Added annotations
- Removed explicit sequence generation
- Moved out from IdentifiableEntity so ID column in DB has clearer name

Progress on #36

### CODE REVIEW

File: psm-app/db/seed.sql [hunk 1]
```diff
@@ -1031,3 +1034,13 @@ CREATE TABLE designated_contacts(
...
 );
+
+  CREATE TABLE accepted_agreements
+   (
+    accepted_agreement_id BIGINT PRIMARY KEY,
+    profile_id BIGINT,
+    ticket_id BIGINT,
+    accepted_date DATE,
+    agreement_document_id BIGINT
+   ) ;
```
Comment (jasonaowen) [inline psm-app/db/seed.sql]: This is indented differently than the rest of the file, could you please fix?

File: psm-app/services/src/main/java/gov/medicaid/entities/AcceptedAgreements.java [hunk 2]
```diff
@@ -15,110 +15,73 @@
...
- * Represents user accepted agreements.
- *
- * @author TCSASSEMBLER
- * @version 1.0
- */
-public class AcceptedAgreements extends IdentifiableEntity {
+@javax.persistence.Entity
+@Table(name = "accepted_agreements")
+public class AcceptedAgreements implements Serializable {
+ @Id
```
Comment (jasonaowen) [inline psm-app/services/src/main/java/gov/medicaid/entities/AcceptedAgreements.java]: Inconsistent indent - can you run IntelliJ's Format Code command on this file, please?

File: psm-app/db/seed.sql [hunk 3]
```diff
@@ -1031,3 +1034,13 @@ CREATE TABLE designated_contacts(
...
     REFERENCES people(entity_id)
 );
+
+  CREATE TABLE accepted_agreements
+   (
+    accepted_agreement_id BIGINT PRIMARY KEY,
+    profile_id BIGINT,
+    ticket_id BIGINT,
+    accepted_date DATE,
+    agreement_document_id BIGINT
```
Comment (jasonaowen) [inline psm-app/db/seed.sql]: This should have a foreign key to `agreement_documents`.

File: psm-app/db/seed.sql [hunk 4]
```diff
@@ -1,7 +1,9 @@
 DROP SEQUENCE IF EXISTS
+DROP SEQUENCE IF EXISTS
```
Comment (cecilia-donnelly) [inline psm-app/db/seed.sql]: Is this a duplicate line?

File: psm-app/services/src/main/java/gov/medicaid/services/util/Sequences.java [hunk 5]
```diff
@@ -90,14 +85,14 @@
     public static final String PAY_TO_SEQ = "PAY_TO_SEQ";
 
     public static final String SERVICE_SEQ = "SERVICE_SEQ";
-    
+
```
Comment (cecilia-donnelly) [inline psm-app/services/src/main/java/gov/medicaid/services/util/Sequences.java]: Try to minimize these extra whitespace changes where possible -- they add some noise to the diff.

**Comment by cecilia-donnelly** [comment]
@BenGalewsky just add `AcceptedAgreements` to persistence.xml and this should be good to go.  Thanks!

File: psm-app/db/seed.sql [hunk 6]
```diff
@@ -1031,3 +1033,12 @@ CREATE TABLE designated_contacts(
   person_id BIGINT
     REFERENCES people(entity_id)
 );
+
+CREATE TABLE accepted_agreements(
+  accepted_agreement_id BIGINT PRIMARY KEY,
+  profile_id BIGINT,
+  ticket_id BIGINT,
+  accepted_date DATE,
+  agreement_document_id BIGINT REFERENCES  documents(document_id)
```
Comment (jasonaowen) [inline psm-app/db/seed.sql]: This should reference `agreement_documents`, not `documents`.

One thing I do to double-check my work is to turn on Hibernate's schema autocreate feature, don't run `seed.sql`, deploy the app, and examine the schema it creates.

**Pull request merged**